src/backend/cuda/platform.cpp

```diff
@@ -37,7 +39,18 @@
 
 using namespace std;
 
+
 namespace cuda {
+
+// check with reviewers : perhaps to be moved to a static of DeviceManager
+std::shared_ptr<spdlog::logger> logger = common::loggerFactory("cuda");
```

Comment:
Your comment is right. This logger should be defined as member of CUDA backend's device manger class.

You can refer to `src/backend/common/MemoryManager.hpp` and `src/backend/common/MemoryManagerImpl.hpp` to see how the `mem` logger is tied to the memory manager's context.

Similarly, we should keep CUDA backend logger tied to the CUDA device manager instead of having a global variable.

Reply:
Forward declaring `spdlog::logger` inside `src/backend/cuda/platform.hpp` is sufficient to declare a shared_ptr of logger inside DeviceManager class. This forward declaration is actually already present in `src/backend/cuda/platform.hpp`, so you can just move the declaration to class interior and initialization to the class's constructor.

Reply:
Not fan as it would prevent to use the logger without the DeviceManager but anyway, done.

Reply:
The idea is to not have a single global logger with all the message clubbed into one group but instead have loggers tied to specific contexts like we have `mem` & `unified` for memory-manager and unified-api, respectively.